.ci/releasePipeline.groovy

```diff
@@ -141,3 +65,161 @@ pipeline {
         }
     }
 }
+
+
+String getVersionFromGitTagName() {
+    def ver = sh(returnStdout: true, script: "git tag -l --points-at HEAD --sort=-creatordate | head -n 1")
+        .trim()
+        .substring(1)  // drop v prefix
+    echo "version= ${ver}"
+    return ver
+}
+
+void getSignClient() {
+    withVault(vaultSecrets: [[path        : 'keptn-jenkins/monaco/sign-service',
+                              secretValues: [[envVar: 'repository', vaultKey: 'client_repository_url ', isRequired: true]]]]
+    ) {
+        if (!fileExists('sign-client.jar')) {
+            sh 'curl -Ss -g -o sign-client.jar $repository/testautomation-release-local/com/dynatrace/services/signservice/client/1.0.[RELEASE]/client-1.0.[RELEASE].jar'
+        }
+    }
+}
+
+void buildBinary(Map args = [binary: '', version: '', ctx: null]) {
+    args.ctx = args.ctx ?: ctx
+    stage('🏁 Build ' + args.binary) {
+        sh "make ${args.binary} VERSION=${args.version} OUTDIR=${args.ctx.unsignedDir}"
+    }
+}
+
+void signBinaries(Map args = [binary: '', version: '', ctx: null]) {
+    args.ctx = args.ctx ?: ctx
+    stage('Sign ' + args.binary) {
+        script {
+            switch (getOs(args.binary)) {
+                case "windows":
+                    signWinBinaries(args)
+                    break
+                case "linux":
+                case "darwin":
+                    signLinuxBinaries(args)
+                    break
+            }
+        }
+    }
+}
+
+void signWinBinaries(Map args) {
+    withEnv(["binarySource=${args.ctx.unsignedDir}/${args.binary}", "version=${args.version}", "destinationDir=${args.ctx.signedDir}"]) {
+        withVault(vaultSecrets: [[path        : 'signing-service-authentication/monaco',
+                                  secretValues: [
+                                      [envVar: 'username', vaultKey: 'username', isRequired: true],
+                                      [envVar: 'password', vaultKey: 'password', isRequired: true]]]]
+        ) {
+            sh '''java -jar sign-client.jar \
+                        --action SIGN \
+                        --digestAlg SHA256 \
+                        --downloadPath $destinationDir \
+                        --kernelMode false  \
+                        --triggeringProject monaco \
+                        --triggeringVersion $version \
+                        --username $username \
+                        --password $password \
+                        $binarySource'''
+        }
+    }
+}
+
+void signLinuxBinaries(Map args) {
+    def shaSumFile = "SHA256SUMS"
```

Comment:
as we create a file per binary I would suggest to call it by the binaries name
```suggestion
    def shaSumFile = "${args.binary}.sha"
```

A SHASUMS file like e.g. used by Ubuntu makes more sense if the file contains the sums for several files